commons/src/connections/threads.ts

```diff
@@ -17,6 +17,33 @@ export const fetchThreads = (
   limit: number,
   offset: number,
 ): Promise<Thread[]> => {
+  if (query.thread_ids) {
+    // &limit=${limit}&offset=${offset}
+    const page_of_ids = query.thread_ids.slice(offset, offset + limit);
+    return Promise.all(
+      page_of_ids.map(async (thread_id) => {
+        // Nylas API does not support fetching threads in bulk, so must fetch them one
+        // at a time :(
+        const queryString = `${getMiddlewareApiUrl(
+          query.component_id,
+        )}/threads/${thread_id}?view=expanded`;
```

Comment:
It would be a lot faster if the Nylas API had a way to fetch a list of threads in a single call. Also, fetching the expanded view seems like a lot of data. Combined, these queries are slow and would ideally be improved.